### PR TITLE
Non boolean expectation - expect_enum

### PR DESCRIPTION
**Pull request by arbulu89**
New non boolean expectation implementation: `expect_enum`. Honestly, I have given a lot of thoughts about this name, and even though I'm not 100% satisfied, it is the best that I got. Open for proposals.

This new `expect_enum` expectation simply expects a `passing|warning|critical` return value. If non of them is returned, `critical` will be the value.

It doesn't have much more. The `return_value` of this expectation is again `passing|warning|critical` and the final result is aggregated based on this value.

It includes a `warning_message` new field, used to interpolate an error message if the return_value is `warning`.

PD: I will send other PR with openapi schemas, etc

This is how it looks like in the API:
image

### CODE REVIEW

**Review by EMaksy — APPROVED** [review]
LGTM!

File: lib/wanda/executions/evaluation.ex [hunk 1]
```diff
@@ -206,49 +208,109 @@ defmodule Wanda.Executions.Evaluation do
...
+         %ExpectationEvaluation{type: :expect_enum, return_value: :passing} =
+           expectation_evaluation,
+         _,
+         _,
+         _evaluation_scope,
+         _
+       ),
+       do: expectation_evaluation
+
+  defp maybe_add_failure_message(
```
Comment (CDimonaco) [inline lib/wanda/executions/evaluation.ex:250]: Could we have this at the end of `maybe_add_failure_message` expect_enum clauses?, so we have expect enum with passing, critical and warning, and then this "default"

Comment (arbulu89) [reply lib/wanda/executions/evaluation.ex:250]: we cannot really do as the next 2 functions would override this all the times.
We need to check for the `nil` values.
But let me have a look on that. To see if I can improve it somehow (but i'm not sure I can)

File: lib/wanda/executions/evaluation.ex [hunk 2]
```diff
@@ -327,19 +397,26 @@ defmodule Wanda.Executions.Evaluation do
...
-           not errors?(agents_check_results) do
-        :passing
-      else
+      if errors?(agents_check_results) do
         Enum.find_value(agents_check_results, severity, fn
           %AgentCheckError{type: :timeout} -> :critical
           _ -> false
         end)
+      else
+        expectation_results
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply lib/wanda/executions/evaluation.ex:406]: something like: `# Normalize the expectation results to passing/warning/critical and weight them` ?

[the remark this answers is not in the document]
Comment (arbulu89) [reply lib/wanda/executions/evaluation.ex:412]: Done

**Pull request merged**